MerlinAU.sh

```diff
@@ -806,15 +806,18 @@ _GetAllNodeSettings_()
     [ ! -d "$SETTINGS_DIR" ] && mkdir -m 755 -p "$SETTINGS_DIR"
 
     if [ -f "$SETTINGSFILE" ]; then
-        # Construct the setting key pattern to match
-        pattern="${node_mac_address}.*${setting_part}"
+        # Escape characters that might be interpreted by regex engines, like ':' in MAC addresses
+        local escaped_mac_address=$(echo "$node_mac_address" | sed 's/[\*\.]/\\&/g')
+
+        # Construct the pattern to match the entire line containing the MAC address and the setting part
+        pattern="${escaped_mac_address}.*${setting_part}=\"[^\"]*\""
```

Comment:
The regex seems to be more complicated than it needs to be, but perhaps I'm missing something because I don't have any AiMesh nodes so I cannot really recreate the problem (I didn't fully understand the old code either because of the same issue: no mesh nodes).

What's the exact key/value pattern that you're trying to find/match in the settings file? 
Can you give me a couple of examples from your settings file (just obfuscate your MAC addresses)?

I think I need more context to understand what you're trying to do here so I can wrap my head around it.

Reply:
@Martinski4GitHub 

Sorry for the delay, I had other things come up that took priority. I know you understand.
I did read the message but wasn't in a situation were I could respond at the moment.

At your request is is the contents of my settings file:

```
Node_7C:XX:XX:XX:XX:XX_Model_NameID="GT-AXE11000-DBF0"
Node_7C:XX:XX:XX:XX:XX_New_Notification_Date="2024-04-26_19:22:00"
Node_7C:XX:XX:XX:XX:XX_New_Notification_Vers="3004.388.7.0"
```

![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/d1113e10-a903-401e-b94e-f1880b0672ca)

Reply:
And every MAC address is the same for the one node I'm trying to look up.

Reply:
Resolved in commit: https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/197/commits/48349af30f7310ef9f8cb0e71d8cf2cd7075191f

Reply:
> 
> At your request is is the contents of my settings file:
> 
> ```
> Node_7C:XX:XX:XX:XX:XX_Model_NameID="GT-AXE11000-DBF0"
> Node_7C:XX:XX:XX:XX:XX_New_Notification_Date="2024-04-26_19:22:00"
> Node_7C:XX:XX:XX:XX:XX_New_Notification_Vers="3004.388.7.0"
> ```
>
Thank you. This provides the necessary context to understand what your code is trying to do. With this in mind, now I  can review & follow the code to try to wrap my head around it.